api/v1alpha1/oauth2client_types.go

```diff
@@ -0,0 +1,84 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OAuth2ClientSpec defines the desired state of OAuth2Client
+type OAuth2ClientSpec struct {
+	// GrantTypes is an array of grant types the client is allowed to use.
+	//
+	// Pattern: client_credentials|authorization_code|implicit|refresh_token
```

Comment:
If we already have this so well defined, maybe use the kubebuilder [verification annotations](https://book.kubebuilder.io/reference/markers/crd-validation.html) to include this in the CRD?

Reply:
validation added!